### PR TITLE
add: implement docker e2e ci job

### PR DESCRIPTION
**Pull request by marcello33**
# Description

Implement the docker e2e ci job the same way it is done in `bor` and `heimdall`, to make sure the docker setup always work as expected. This also ensures `matic-cli` changes do not affect `bor` and `heimdall` CIs.
The CI now tests:
- `express-cli` in a remote environment on AWS
- `matic-cli` in a dockerized environment locally

The PR also merges a dependabot PR (see https://github.com/maticnetwork/matic-cli/pull/219)

# Changes

- [ ] Bugfix (non-breaking change that solves an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)
- [ ] New test case for remote devnet

# Checklist

- [x] I have added at least 2 reviewer or the whole pos-v1 team
- [ ] I have added sufficient documentation in code
- [x] I will be resolving comments - if any - by pushing each fix in a separate commit and linking the commit hash in the comment reply

## Testing

- [x] I have tested this code manually on local environment
- [ ] I have tested this code manually on remote devnet using express-cli
- [ ] I have tested this code manually on mumbai

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -134,3 +134,90 @@ jobs:
...
+        with:
+          node-version: ${{ env.NODE_VERSION }}
+
+      - name: Install npm dependencies
+        working-directory: matic-cli
+        run: npm install --prefer-offline --no-audit --progress=false
+
+      - name: Bootstrap devnet
+        run: |
+          cd matic-cli
```
Comment (Raneet10) [inline .github/workflows/ci.yml:173]: nit: maybe use `working-directory` like above ?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -134,3 +134,90 @@ jobs:
...
+          npm install --prefer-offline --no-audit --progress=false
```
Comment (Raneet10) [inline .github/workflows/ci.yml:174]: We are installing dependencies again after the `Install npm dependencies` job ?

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -134,3 +134,90 @@ jobs:
...
+          mkdir devnet
+          cd devnet
+          ../bin/matic-cli.js setup devnet -c ../../matic-cli/.github/e2e-docker-config.yml
+
+      - name: Launch devnet
+        run: |
+          cd matic-cli/devnet
```
Comment (Raneet10) [inline .github/workflows/ci.yml:181]: could use `working-directory` keyword.

Comment (Raneet10) [reply .github/workflows/ci.yml:181]: Or are we avoiding because we have to change directories several times ?

**Review by Raneet10 — APPROVED** [review]
Minor comments, else LGTM

File: .github/workflows/ci.yml (hunk 1 above)
Comment (marcello33) [reply .github/workflows/ci.yml]: Agree, done [here](9f2e436179eaa363000a9dc40fe931a2353fa68a)

File: .github/workflows/ci.yml (hunk 2 above)
Comment (marcello33) [reply .github/workflows/ci.yml]: Good catch, thanks. Removed [here](9f2e436179eaa363000a9dc40fe931a2353fa68a)

File: .github/workflows/ci.yml (hunk 3 above)
Comment (marcello33) [reply .github/workflows/ci.yml]: No, just copied from bor, but good suggestion.
Applied [here](9f2e436179eaa363000a9dc40fe931a2353fa68a)

**Comment by marcello33** [comment]
> Minor comments, else LGTM

Thanks for the review @Raneet10 
Applied some changes. Will wait for the CI to complete.

**Comment by marcello33** [comment]
@Raneet10 had to push [this]([f311935](https://github.com/maticnetwork/matic-cli/pull/222/commits/f311935b9549ed80406fdbeea3ff65184f5e7ac9)) due to latest changes.
Let's see if this makes the e22 docker tests pass

**Review by manav2401 — APPROVED** [review]

**Pull request merged**